topo_processor/util/aws_credentials.py

```diff
@@ -1,29 +1,51 @@
+import json
 import os
+from typing import Dict
 
 import boto3
 
-default_credentials = {}
+from topo_processor.file_system.get_fs import bucket_name_from_path
+from topo_processor.util.configuration import aws_role_config_path
+
+
+class Credentials:
+    access_key: str
+    secret_key: str
+    token: str
+
+    def __init__(self, access_key: str, secret_key: str, token: str):
+        self.access_key = access_key
+        self.secret_key = secret_key
+        self.token = token
+
+
+default_credentials: Credentials
 session = boto3.Session(profile_name=os.getenv("AWS_PROFILE"))
 client = session.client("sts")
 bucket_roles = {}
 
 
-def get_credentials(bucket_name: str):
+def get_credentials(bucket_name: str) -> Credentials:
+    if not bucket_roles:
+        load_roles(json.load(open(aws_role_config_path)))
 
     if bucket_name in bucket_roles:
         if not bucket_roles[bucket_name]["client"]:
             client.assume_role(RoleArn=bucket_roles[bucket_name]["role"])
             bucket_roles[bucket_name]["client"] = client
         cred = bucket_roles[bucket_name]["client"].get("Credentials")
-        return {
-            "access_key": cred.get("AccessKeyId"),
-            "secret_key": cred.get("SecretAccessKey"),
-            "token": cred.get("SessionToken"),
-        }
+
+        return Credentials(cred.get("AccessKeyId"), cred.get("SecretAccessKey"), cred.get("SessionToken"))
 
     if not default_credentials:
         session_credentials = session.get_credentials()
-        default_credentials["access_key"] = session_credentials.access_key
-        default_credentials["secret_key"] = session_credentials.secret_key
-        default_credentials["token"] = session_credentials.token
+        default_credentials = Credentials(
+            session_credentials.access_key, session_credentials.secret_key, session_credentials.token
+        )
+
     return default_credentials
+
+
+def load_roles(roles_config: Dict) -> None:
+    for (key, value) in roles_config.items():
+        bucket_roles[bucket_name_from_path(key)] = {"role": value["roleArn"]}
```

Comment:
might be best to make this the same. eg use 

```
bucket_roles[bucket_name_from_path(key)] = value
```

so it's easy to reason about.